include/RAJA/exec-cuda/forall_cuda.hxx

```diff
@@ -709,18 +866,23 @@ template <size_t BLOCK_SIZE, typename LOOP_BODY>
 RAJA_INLINE void forall_Icount(
     IndexSet::ExecPolicy<seq_segit, cuda_exec<BLOCK_SIZE>>,
     const IndexSet& iset,
-    LOOP_BODY loop_body)
+    LOOP_BODY&& loop_body)
 {
+  beforeCudaKernelLaunch();
+
+  auto body = loop_body;
+
   int num_seg = iset.getNumSegments();
   for (int isi = 0; isi < num_seg; ++isi) {
     const IndexSetSegInfo* seg_info = iset.getSegmentInfo(isi);
-    executeRangeList_forall_Icount<cuda_exec_async<BLOCK_SIZE>>(seg_info,
-                                                                loop_body);
+    executeRangeList_forall_Icount<cuda_exec_async<BLOCK_SIZE>>(seg_info, body);
 
   }  // iterate over segments of index set
 
   cudaErrchk(cudaPeekAtLastError());
   cudaErrchk(cudaDeviceSynchronize());
+
+  afterCudaKernelLaunch();
 }
 
```

Comment:
These functions ultimately call back into the other cuda forall launcher functions, so they do not need to call before/afterCudaKernelLaunch or to copy the loop_body. In fact it would be better to pull those calls up into this function entirely instead of calling them for each function call. Also is there a need for these functions to be always synchronous?